File: BitBoy/app/src/main/java/lib/core/ViewCore.java
package lib.core;

import lib.CustomTimer;
import lib.CustomTimer.TimerDelegate;
import lib.ViewUtil;

import android.content.Context;
import android.content.Intent;
import android.content.res.Configuration;
import android.util.Log;
import android.view.View;
import android.view.animation.Animation.AnimationListener;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.Toast;

import com.ironraft.bitboy.R;


public class ViewCore extends FrameLayout implements  TimerDelegate
{
    protected Button btnBack;
	protected ActivityCore mainActivity;
	private Boolean isMoveInit;
	private CustomTimer resizeTimer;
	protected Boolean isDirectBack;
	public PageObject pageInfo;


	public ViewCore(Context context,PageObject _pageInfo) { 
        super(context);
        pageInfo=_pageInfo;
        isDirectBack=true;
        isMoveInit=false;
        mainActivity= ActivityCore.getInstence();
        this.setClickable(true);
       // View.inflate(context, R.layout.view_main,this);


    } 
	protected void setResizeHendler() { 
		removeResizeTimer();
		resizeTimer=new  CustomTimer(100,1,this);
		resizeTimer.resetTimer();
    } 
	
	public void onTime(CustomTimer timer){
		
	}
	public void onComplete(CustomTimer timer){
    	resize();
    }
	protected void removeResizeTimer() { 
		if(resizeTimer!=null){
			resizeTimer.removeTimer();
			resizeTimer=null;
		}
    }
    public void initCore() {
        //btnBack=(Button) findViewById(R.id._btnBack);

        doInit();
    }
    public void movedInit() {
	    if(isMoveInit){
	    	return;
	    }
       // btnBack=(Button) findViewById(R.id._btnBack);
	    isMoveInit=true;

        if(btnBack!=null) {
            btnBack.setOnClickListener(new OnClickListener() {

                @Override
                public void onClick(View v) {
                    ActivityCore.getInstence().changeViewBack();
                }
            });
        }
        doMovedInit();
    } 
	

    public Boolean directBack() { 
    	
    	doDirectBack();
    	return isDirectBack;
    } 
	public void activityResult(int requestCode, Intent data) { 
    	
		doActivityResult(requestCode, data);
    }

    public void configurationChanged(Configuration newConfig){
    	
		doConfigurationChanged(newConfig);
		if(resizeTimer!=null)
		{
			resizeTimer.timerStop();
			resizeTimer.resetTimer();
			resizeTimer.timerStart();
		}
    }
    public void setBodySizeChanged(int proposedWidth,int proposedheight)
    {
        doSetBodySizeChanged(proposedWidth, proposedheight);
    }
    public void removeViewInit() {


        doRemoveInit();

    }
    public void removeViewCore() { 
    	
    	removeResizeTimer();
    	doRemove();
    	mainActivity=null;
    }

    public void resize(){ 
    	int w= this.getWidth();
    	int h= this.getHeight();
       // Log.i("", "ViewCore Resize w : " + w + " h : " + h);
    	doResize();
    }

    public void onUpdate()
    {
        doUpdate();
    }
    public void onPause(){ 
    	
    	doPause();
    } 
    public void onResume(){ 
    	
    	doResume();
    }

    protected void doUpdate() {

        // Log.i("","ViewCore Resize!!");
    }
    protected void doResize() { 
    	 
    	// Log.i("","ViewCore Resize!!");
    }
    protected void doSetBodySizeChanged(int proposedWidth,int proposedheight) {

    }
    protected void doActivityResult(int requestCode, Intent data) { 
        
    }
    protected void doConfigurationChanged(Configuration newConfig) {
        
    }
    protected void doInit() {

    }
    protected void doMovedInit() { 
        
    }
    protected void doDirectBack() { 
        
    }
    protected void doRemoveInit() {
        
    }
    protected void doRemove() {

    }
    protected void doPause() { 
        
    } 
    protected void doResume() { 
    
    } 
} 

File: Magellan/app/src/main/java/lib/core/ViewCore.kt
package lib.core

import android.content.Context
import android.content.Intent
import android.content.res.Configuration
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import lib.CustomTimer

open class ViewCore(context: Context, pageInfo: PageObject = PageObject(0)) : FrameLayout(context), CustomTimer.TimerDelegate
{
    open var btnBack: Button? = null
    open var mainActivity: ActivityCore? = null
    private var isMoveInit: Boolean = false
    private var resizeTimer: CustomTimer? = null
    open var isDirectBack: Boolean = true
    var pageInfo: PageObject = pageInfo

    init
    {
        mainActivity = ActivityCore.shareInstence()
        this.isClickable = true
        // View.inflate(context, R.layout.view_main,this);

    }

    open fun setResizeHendler()
    {
        removeResizeTimer()
        resizeTimer = CustomTimer(100, 1, this)
        resizeTimer!!.resetTimer()
    }

    override fun onComplete(timer: CustomTimer)
    {
        resize()
    }

    open fun removeResizeTimer()
    {
        resizeTimer?.removeTimer()
        resizeTimer = null
    }

    fun initCore()
    {
        //btnBack=(Button) findViewById(R.id._btnBack);
        doInit()
    }

    fun movedInit()
    {
        if (isMoveInit == true) return
        isMoveInit = true
        btnBack?.setOnClickListener { ActivityCore.shareInstence()?.changeViewBack() }
        doMovedInit()
    }


    fun directBack(): Boolean
    {
        doDirectBack()
        return isDirectBack
    }

    fun activityResult(requestCode: Int, data: Intent) {

        doActivityResult(requestCode, data)
    }

    fun configurationChanged(newConfig: Configuration) {

        doConfigurationChanged(newConfig)
        resizeTimer?.timerStop()
        resizeTimer?.resetTimer()
        resizeTimer?.timerStart()
    }

    open fun setBodySizeChanged(proposedWidth: Int, proposedheight: Int) {
        doSetBodySizeChanged(proposedWidth, proposedheight)
    }

    fun removeViewInit()
    {
        doRemoveInit()
    }

    fun removeViewCore()
    {
        removeResizeTimer()
        doRemove()
        mainActivity = null
    }

    fun resize()
    {
        //val w = this.width
        //val h = this.height
        // Log.i("", "ViewCore Resize w : " + w + " h : " + h);
        doResize()
    }

    fun onUpdate()
    {
        doUpdate()
    }

    fun onPause()
    {
        doPause()
    }

    fun onResume()
    {
        doResume()
    }

    open fun doUpdate() {

        // Log.i("","ViewCore Resize!!");
    }

    open fun doResize() {

        // Log.i("","ViewCore Resize!!");
    }

    open fun doSetBodySizeChanged(proposedWidth: Int, proposedheight: Int) {

    }

    open fun doActivityResult(requestCode: Int, data: Intent) {

    }

    open fun doConfigurationChanged(newConfig: Configuration) {

    }

    open fun doInit() {

    }

    open fun doMovedInit() {

    }

    open fun doDirectBack() {

    }

    open fun doRemoveInit() {

    }

    open fun doRemove() {

    }

    open fun doPause() {

    }

    open fun doResume() {

    }
}